subsys/net/ip/dhcpv4.c

```diff
@@ -1236,6 +1276,30 @@ static void dhcpv4_start_internal(struct net_if *iface, bool first_start)
 	k_mutex_unlock(&lock);
 }
 
+int net_dhcpv4_add_option_callback(struct net_dhcpv4_option_callback *cb)
+{
+	__ASSERT(cb, "No callback!");
+	__ASSERT(cb->handler, "No callback handler!");
+
+	sys_slist_prepend(&option_callbacks, &cb->node);
```

Comment:
`sys_slist` is not thread safe, so any operations on it should be protected. I think `lock` mutex defined in the file could be used here (it's already locked for instance when we loop over the callbacks above).

Reply:
Ok, will be fixed